framework/src/base/FEProblem.C

```diff
@@ -416,7 +416,7 @@ void FEProblem::initialSetup()
     Moose::setup_perf_log.push("initial adaptivity", "Setup");
     unsigned int n = adaptivity().getInitialSteps();
 
-    if (n && !_app.isUltimateMaster())
+    if (n && !_app.isUltimateMaster() && _app.isRestarting())
```

Comment:
Yes this fixes a bug fix, the MultiApps that have sub-apps with initial adaptivity will not run because this error is thrown.